sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -387,9 +298,17 @@ export class MessageSender extends LinkEntity {
               this.name,
               delivery.id
             );
+            return resolve();
           } catch (error) {
-            removeListeners();
+            error = translate(error);
```

Comment:
I believe we need to check if there is an `innerError` property on error first, because otherwise we'll always treat the error as retryable:
https://github.com/Azure/azure-sdk-for-js/issues/3375#issuecomment-517844630

Reply:
Interesting.. updated here.

Reply:
Are there any other places in the SDK where we need to look for innerError property that we might have missed?

Reply:
That's the only one I know of. It's the only Error type from rhea-promise that has an innerError anyway.